phf_macros/src/lib.rs

```diff
@@ -225,34 +227,40 @@ fn parse_set(cx: &mut ExtCtxt, tts: &[TokenTree]) -> Option<Vec<Entry>> {
     Some(entries)
 }
 
-fn parse_key(cx: &mut ExtCtxt, e: &Expr) -> Option<Key> {
+fn parse_key(cx: &mut ExtCtxt, e: P<Expr>) -> Option<(Key, P<Expr>)> {
```

Comment:
Could we keep the signature of this function the same and add a new function that adjusts the expr? That will let us avoid all of the `e.clone()`s.